core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -64,6 +64,19 @@ private[log] class LogCleanerManager(val logDirs: Array[File], val logs: Pool[To
   @volatile private var dirtiestLogCleanableRatio = 0.0
   newGauge("max-dirty-percent", new Gauge[Int] { def value = (100 * dirtiestLogCleanableRatio).toInt })
 
+  /* a gauge for tracking the number of logCleaner runs */
+  @volatile private var numLogCleanRuns = 0
+  newGauge("num-runs", new Gauge[Int] { def value = numLogCleanRuns })
+
+  /* a gauge for tracking the time since the last log cleaner run, in milli seconds */
+  @volatile private var timeSinceLastRun : Long = System.currentTimeMillis
+  newGauge("time-since-last-run-ms", new Gauge[Long] { def value = System.currentTimeMillis - timeSinceLastRun })
+
+  /* a gauge for tracking the number of filthy logs. A non zero value indicates
+   * that the cleaner has not been successful in cleaning the logs. */
+  @volatile private var numFilthyLogs  = 0
+  newGauge("num-filthy-logs", new Gauge[Int] { def value = numFilthyLogs })
```

Comment:
The other two metrics (num-runs and last-run-time) tell us if log cleaner has been active or not. However, I think there is a possibility of a log remaining in uncleaned state even when log cleaner is active. The clean function in LogCleaner.scala, line 237 could throw an exception.  The “doneCleaning” function in finally block would update the checkpoints and remove it from inProgress hash map. (Meaning, log is not cleaned, right?). "num-filthy-logs" would catch this case, i.e logs that were supposed to be cleaned, but did not get cleaned.